parsing/pcoq.ml

```diff
@@ -411,6 +411,8 @@ module Prim =
     let ne_string = Entry.create "Prim.ne_string"
     let ne_lstring = Entry.create "Prim.ne_lstring"
 
+    let bar_cbrace = Entry.create "'|}'"
```

Comment:
I did initially use an alphabetic name, but then I realised the main use of that string is in printing diagnostics, and I think the message `expected ['|}'] after [constr:record_declaration]` was easier to act upon than `expected [prim:bar_right_curly_brace] after [constr:record_declaration]` (Ideally, I'd have liked to remove the enclosing `[]`.)

Reply:
Good remark. 

Maybe can we even amend `gramlib` so that it displays in this case `'|}'` rather than `['|}']`.

Also, if someone has time for it, that's probably worth to eventually make another general pass on entry names so as to make them more appealing. For instance, the name `constr:operconstr` looks a bit frightening and low-level to me (why not simply `constr`?). Maybe something to be done jointly with #9884, with an hyperlink to the reference manual in the error message. (Or am I unrealistic?)

Reply:
I agree `constr:operconstr` looks complicated.  I've not seen any explanation of non-terminal prefixes in the doc, which AFAIK doesn't use them anywhere.  It's not clear how useful this is to non-developer users.  Adding a hyperlink as you suggest may be OK, but hyperlink names are determined while building the doc.  Putting the hyperlink in the OCaml code by hand may be fragile.  I think there are quite a few improvements to make related to grammars and their documentation; I'd create a new issue.  (Follow on work after #9884 already suggests significant cleanup of the grammars (without changing what they recognize) as well as revising large swathes of the documentation to show the grammar.)